src/formats/roam-json.ts

```diff
@@ -266,10 +355,48 @@ export class RoamJSONImporter extends FormatImporter {
 
 		if (json.children) {
 			for (const child of json.children) {
-				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true));
+				markdown.push(await this.jsonToMarkdown(graphFolder, attachmentsFolder, child, indent + '  ', true, '', this.oldestTimestamp, this.newestTimestamp));
+			}
+		}
+
+		// once processing children is completed, add the YAML to the top
+		// check if any YAML options are set
+		if (this.fileDateYAML || this.titleYAML) {
+			checkYAMLoptions = true;
+		}
+
+		// add YAML frontmatter, if enabled
+		// only run on the initial set, skip if child 
+		if (checkYAMLoptions && !isChild) {
```

Comment:
Can we remove `checkYAMLoptions` entirely?

```suggestion
		if ((this.fileDataYAML || this.titleYAML) && !isChild) {
```

Reply:
Nice, thanks! Yeah, as long as there aren't too many options added (potentially) that would make that confusing, this is a better approach